ext/re2/extconf.rb

```diff
@@ -8,41 +8,79 @@
 
 incl, lib = dir_config("re2", "/usr/local/include", "/usr/local/lib")
 
-$CFLAGS << " -Wall -Wextra -funroll-loops"
+class CXXFlags
+  def <<(extra_flags)
+    # Newer mkmf
+    if defined? $CXXFLAGS
```

Comment:
For posterity: looks like`$CXXFLAGS` was introduced in Ruby 2.2 in https://github.com/ruby/ruby/commit/e8792774dff09386a930f4a93fa6fdc864b2e720